libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -632,29 +611,35 @@ private Optional<JsonNode> generatePassportClaim(List<VerifiableCredential> vcs)
             return Optional.empty();
         }
 
-        var passportNode =
-                extractSubjectDetailFromVc(
-                        PASSPORT_PROPERTY_NAME,
-                        passportVc.get(),
-                        "Error while parsing Passport CRI credential",
-                        ErrorResponse.FAILED_TO_GENERATE_PASSPORT_CLAIM);
+        if (passportVc.get().getCredential()
+                instanceof IdentityCheckCredential identityCheckCredential) {
 
-        if (passportNode.isMissingNode()) {
-            StringMapMessage mapMessage =
-                    new StringMapMessage()
-                            .with(
-                                    LOG_MESSAGE_DESCRIPTION.getFieldName(),
-                                    "Passport property is missing from VC")
-                            .with(LOG_CRI_ISSUER.getFieldName(), passportVc.get().getCri().getId());
-            LOGGER.warn(mapMessage);
+            var credentialSubject = getIdentityCheckSubjectOrThrowError(identityCheckCredential);
 
-            return Optional.empty();
-        }
+            var passport = credentialSubject.getPassport();
+
+            if (isNullOrEmpty(passport)) {
+                StringMapMessage mapMessage =
+                        new StringMapMessage()
+                                .with(
+                                        LOG_MESSAGE_DESCRIPTION.getFieldName(),
+                                        "Passport property is missing from VC or empty passport property.")
+                                .with(
+                                        LOG_CRI_ISSUER.getFieldName(),
+                                        passportVc.get().getCri().getId());
+                LOGGER.warn(mapMessage);
+                return Optional.empty();
+            }
 
-        return Optional.of(passportNode);
+            return Optional.of(passport);
+        } else {
+            LOGGER.error(LogHelper.buildLogMessage(MUST_BE_IDENTITYCHECK_MESSAGE));
+            throw new HttpResponseExceptionWithErrorBody(
+                    500, ErrorResponse.FAILED_TO_GENERATE_PASSPORT_CLAIM);
```

Comment:
Sorry one small thing - I think these all used to be warnings, but now they are errors...

Can we turn this into a warning log and return `Optional.empty` instead?